cloud/services/networkinterfaces/networkinterfaces.go

```diff
@@ -66,14 +66,9 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 
 	nicConfig := &network.InterfaceIPConfigurationPropertiesFormat{}
 
-	subnetInterface, err := subnets.NewService(s.Scope).Get(ctx, &subnets.Spec{Name: nicSpec.SubnetName, VnetName: nicSpec.VnetName})
+	subnet, err := subnets.NewService(s.Scope).Client.Get(ctx, s.Scope.Vnet().ResourceGroup, nicSpec.VnetName, nicSpec.SubnetName, "")
```

Comment:
Consider putting NewService on its own line. The code is a bit easier to read that way.